src/trajectory_tracking.h

```diff
@@ -0,0 +1,62 @@
+
+#ifndef __TRACY__
+#define __TRACY__
+
+#include <stdint.h>
+
+#define DEFAULT_PARAM_FANCY_GREEK_C (0.7f)
+#define DEFAULT_PARAM_G (60.0f)
+
+/**
+ * output of the linear controller
+ *
+ */
+typedef struct {
+    float rule1;
+    float rule2;
+} feedback_rule_t;
+
+/**
+ * represents a tracking error: reference_pos - current_pos
+ *
+ */
+typedef struct {
+    float x_error;
+    float y_error;
+    float theta_error;
+} tracking_error_t;
+
+/**
+ * velocity of reference in polar robot coordinates
+ *
+ */
+typedef struct {
+    float radial_velocity;
+    float angular_velocity;
```

Comment:
i put radial_velocity for consistency with the reference paper but will change it^^ tangential_velocity makes more sense